lib/src/main/java/growthbook/sdk/java/GBFeaturesRepository.java

```diff
@@ -382,7 +382,13 @@ private void onSuccess(Response response) throws FeatureFetchException {
             ResponseBody responseBody = response.body();
             if (responseBody == null) {
                 throw new FeatureFetchException(
-                    FeatureFetchException.FeatureFetchErrorCode.NO_RESPONSE_ERROR
+                        FeatureFetchException.FeatureFetchErrorCode.NO_RESPONSE_ERROR
+                );
+            }
+            if (response.code() != 200) {
```

Comment:
Hi @micpalmia thanks for your contribution. Unfortunately this approach will be quite fragile and will introduce a regression. In some cases, the API returns 304 with the content, and in those cases, this will break the SDK.

<img width="720" alt="image" src="https://github.com/growthbook/growthbook-sdk-java/assets/113377031/49b491c0-8e00-4ca1-a2ec-5cc0e67c92e1">

Please update your error check to include only error-related status codes.